app/models/admin_notice.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+class AdminNotice < ActiveRecord::Base
+  validates :identifier, presence: true
+
+  enum :priority, %i[low high].freeze
+  enum :category, %i[problem].freeze
+
+  def message
+    I18n.t(
+      "dashboard.#{category}.#{identifier}",
+      **details.symbolize_keys.merge(base_path: Discourse.base_path),
```

Comment:
I think there is a slight bug in this pipeline for cases like `GroupEmailCredentials` that define their own message override with `Problem.new(message)`. I get this on the dashboard and in the `AdminNotice` record:

> Translation missing: en.dashboard.problem.group_email_credentials

I think it has to do with the change here:

https://github.com/discourse/discourse/blob/32b186a2e37b6d54b8cc9250f1c875f0e19aba1a/app/models/problem_check.rb#L143-L150

Also note in this case the **identifier** for each group having a problem is unique because we want to show one per group, and it should be high priority but it's being stored as low priority in the DB:

https://github.com/discourse/discourse/blob/32b186a2e37b6d54b8cc9250f1c875f0e19aba1a/app/services/problem_check/group_email_credentials.rb#L64-L64

I also think we would do well to write a class-level doc inside `ProblemCheck` that does a high level overview of how the system works, e.g.

1. How do ProblemCheck / ProblemCheckTracker / AdminNotice relate to one another and the data flow
1. How to clear out problems manually for retesting
1. How to call/run a problem check manually for debugging (`ProblemCheck::GroupEmailCredentials.run`)

And so on; I think since it will be a complex system it's good to document this lightly from the beginning.

Reply:
@Drenmi I am having a little trouble putting this together in my head still and how the targets fit in, I think a high level flow/architecture documentation comment would still be useful in e.g. the `ProblemCheck` class.

For example I see the `target` is a string in `ProblemCheckTrackers` and we only use the group ID as the target. It's combined with `identifier` so I think it will be good as a secondary uniqueness identifier. I trust that you have all this worked out, so mainly I think for me and for other engineers we just need the high level docs around all this and it will be :ok_hand:

Reply:
I will update the problem check documentation with more information about this. 🙂